### PR TITLE
[WIP] Schedules controller

### PR DESCRIPTION
**Pull request by bdunne**
Add Api::SchedulesController with list and show

Depends on: https://github.com/ManageIQ/manageiq/pull/17561

### CODE REVIEW

**Comment by bdunne** [comment]
@abellotti Please review

File: spec/requests/schedules_spec.rb [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        "name"      => "schedules",
+        "resources" => [{"href" => api_schedule_url(nil, schedule)}]
+      }
+      expect(response).to have_http_status(:ok)
+      expect(response.parsed_body).to include(expected)
+    end
+  end
+
+  context "GET /api/schedules/:id" do
+    it "returns a single Physical Switch" do
```
Comment (abellotti) [inline spec/requests/schedules_spec.rb]: cut & paste ?

File: config/api.yml [hunk 2]
```diff
@@ -2466,12 +2466,25 @@
       - :name: delete
         :identifier: rbac_role_delete
   :schedules:
-    :description: Report schedules
-    :identifier: miq_report_reports
+    :description: Schedules
+    :identifier: schedule
     :options:
+    - :collection
+    - :show
```
Comment (abellotti) [inline config/api.yml]: was the show option needed ?

Comment (bdunne) [reply config/api.yml]: ¯\\\_(ツ)_/¯ The tests pass without it, so I guess I don't know what `show` means. :smile:

File: spec/requests/schedules_spec.rb (hunk 1 above)
Comment (bdunne) [reply spec/requests/schedules_spec.rb]: yes, fixed

**Pull request closed without merging**